sky/engine/core/script/dart_controller.cc

```diff
@@ -156,4 +158,43 @@ void DartController::InstallView(View* view) {
   builtin_sky_->InstallView(view);
 }
 
+static void DartController_DartStreamConsumer(
+    Dart_StreamConsumer_State state,
+    const char* stream_name,
+    uint8_t* buffer,
+    intptr_t buffer_length,
+    mojo::DataPipeProducerHandle* handle) {
+
+  if (handle == nullptr) {
+    // Simple flush. Nothing to do.
+    return;
+  }
+
+  if (state == Dart_StreamConsumer_kData) {
+    uint32_t length = buffer_length;
+    mojo::WriteDataRaw(*handle, buffer, &length,
+                       MOJO_WRITE_DATA_FLAG_ALL_OR_NONE);
+  }
```

Comment:
What if there's not enough room in the pipe?  You generally need a way to buffer on the sending side and then to wait asynchronously on the data pipe have space.

Is the goal to copy all the data into the pipe synchronously or can we buffer it in memory and feed it into the pipe asynchronously?